user_technical/reference_guide/tensorflow.md

```diff
@@ -0,0 +1,207 @@
+## Memgraph Tensorflow Op
+
+### Introduction
+Memgraph Tensorflow Op is Tensorflow component used for interactions with Memgraph database.
+
+### API
+Memgraph Op API consists of inputs, attributes and outputs.
+
+#### Input
+There are two  inputs:
+* query
+* input list
+
+The Query is a string. The query is *openCypher* query supported by Memgraph. Memgraph Op has some limitations on the query. Limitations are explained [here](#limitations).
+
+Input list is a query parameter. Name of this parameter is ```$input_list```. 
+
+Let's see one simple example:
+
+```openCypher
+MATCH (p :Person) WHERE p.id IN $input_list;
+```
+
+Python part:
+
+```python3
+query = "MATCH (p :Person) WHERE p.id IN $input_list;"
+input_list = [1, 2, 3, 4, 5]
+
+with tf.Session() as sess:
+  query_holder = tf.placeholder(tf.string) # Query placeholder
+  input_list_holder = tf.placeholder(tf.int64) # Input list placeholder
+
+  #Create memgraph op
+  memgraph_op = \
+    memgraph_op_module.memgraph_op( \
+    query_holder, \
+    input_list_holder, \
+    output_dtype = tf.int64)
+      
+  # Run tensorflow graph and give input to tensorflow graph
+  sess.run(memgraph_op, {query_holder: query, input_list_holder: input_list})
+```
+
+Query execution replaces ```$input_list``` with provided op input ([see python example for more](#python-example)). Input list is the only **query parameter** used by Memgraph Op.
+
+#### Attributes
+
+Memgraph Op attributes:
+* host, default = 127.0.0.1
+* port, default = 7687
+* user, default = "" (empty string)
+* password = "" (empty string)
+* use_ssl = false
+* output_dtype
+
+*Host*, *port*, *user*, *password* and *use_ssl* are attributes used 
+for connecting to Memgraph. The only different attribute is *output_type*. 
+*output_dtype* has no default value and 
+it is used to determine the type of output tensor. 
+Notice that all data in the output tensor must be of the same type.
+*dtype* can be **int64**, **double**, **bool** and **string**. 
+Memgraph Op does not support other output types.
+
+#### Outputs
+
+Memgraph Op has two outputs:
+* header
+* rows
+
+The header is a string list. The list contains headers provided by query execution:
+
+```openCypher
+MATCH (n) return n.name as Name, n.address as Address;
+```
+
+Header is ["Name", "Address"].
+
+Rows data represent query result. Rows data is the matrix (rows x headers). If there are no results from the query (empty set), the matrix has a dimension (0 x 0).
```

Comment:
represent -> represents the
is the -> is a matrix with dimensions `(|rows| x |headers|)`